src/pykeen/training/training_loop.py

```diff
@@ -502,6 +504,8 @@ def _train(  # noqa: C901
         if not continue_training:
             # Reset the weights
             self.model.reset_parameters_()
+            # afterwards, some parameters may be on the wrong device
+            self.model.to(self.model.get_preferred_device())
```

Comment:
an alternative would be to make sure that in `reset_parameters` the parameters are always re-initialized on the same device. This seemed to be more error-prone and requiring more modifications.